beets/dbcore/query.py

```diff
@@ -553,16 +553,16 @@ def parse(cls, string):
         """
         if not string:
             return None
-        ordinal = string.count('-')
-        if ordinal >= len(cls.date_formats):
-            # Too many components.
-            raise InvalidQueryArgumentTypeError(string,
-                                                'a valid datetime string')
-        date_format = cls.date_formats[ordinal]
-        try:
-            date = datetime.strptime(string, date_format)
-        except ValueError:
-            # Parsing failed.
+        ordinal = 0
+        date = None
+        for date_format in cls.date_formats:
+            try:
+                date = datetime.strptime(string, date_format)
+                break
+            except ValueError:
+                # Parsing failed.
+                ordinal += 1
```

Comment:
Good call. Declaring, looping over, and incrementing `ordinal` on separate lines is not as robust as doing it all on one line.

Come to think of it, the declaration of `ordinal` before the loop wasn't necessary in the first place because in Python loop variables are still in scope after the loop has finished.